### PR TITLE
TIME & SPACE COMPLEXITY ADDED

### PR DESCRIPTION
**Pull request by himanshu14263**

### CODE REVIEW

**Comment by faheel** [comment]
The problem in Kadane's algorithm has already been fixed. See [Kadane.cpp](https://github.com/faheel/Algos/blob/master/MaximumSubarray/Kadane.cpp).

**Comment by faheel** [comment]
The time complexity is O(N) and space complexity is O(1).

**Comment by himanshu14263** [comment]
but i have added the time and space complexity for binary search

File: Searching/BinarySearch.cpp [hunk 1]
```diff
@@ -2,6 +2,9 @@
     Binary search:
     A search algorithm that finds the position of a target value
     within a sorted array
+
+    TIME COMPLEXITY :: O(log n) b/c of recursive implementation.
+    SPACE COMPLEXITY :: O(log n) since in recursion we use stack.
```
Comment (faheel) [inline Searching/BinarySearch.cpp]: Space complexity is `O(1)`.

**Review by faheel — CHANGES_REQUESTED** [review]
The space complexity is incorrect.

**Review by faheel — CHANGES_REQUESTED** [review]
Write it in the format specified in #8.

**Comment by faheel** [comment]
Size (space) of internal stack for function calls isn't considered while calculating the space complexity.

**Comment by faheel** [comment]
@lelouch-lampourge There are a few changes that I've mentioned in the review. Please take a look so that we can merge this as soon as possible.

File: Searching/BinarySearch.cpp [hunk 2]
```diff
@@ -2,6 +2,13 @@
     Binary search:
     A search algorithm that finds the position of a target value
     within a sorted array
+
+    Time complexity:
+    O(log n), where n is the number of elements in the array
```
Comment (faheel) [inline Searching/BinarySearch.cpp:7]: Better write it as:
`O(log(N)), where N ...`

**Review by faheel — CHANGES_REQUESTED** [review]
Your PR also contains the file for Kadane's algorithm, which is not required. Please remove that file.

**Comment by faheel** [comment]
Closing this since there hasn't been any activity for a long time.

**Pull request closed without merging**